staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -58,13 +59,25 @@ type DiffOptions struct {
 	IgnoreChangesAndAdditions bool
 	// IgnoreDeletions indicates if we keep the deletions in the patch.
 	IgnoreDeletions bool
+	// We introduce a new value replaceKeys for patchStrategy.
+	// It indicates that all fields needing to be preserved must be present in the patch.
+	// And the fields that are present will be merged with live object.
+	// All the missing fields will be cleared when patching.
+	ReplaceKeys bool
+	// IgnoreReplaceKeys indicates if we ignore replaceKeys patch strategy.
+	// This is useful to determine if there is a conflict when doing a 3-way diff.
+	// When calculating the diff between original (last-applied) and current (live),
+	// we don't want the patch to respect replaceKeys.
+	IgnoreReplaceKeys bool
 }
 
 type MergeOptions struct {
-	// MergeDeleteList indicates if we are merging the delete parallel list.
-	MergeDeleteList bool
+	// ProcessDirectives indicates if we are merging the delete parallel list.
+	ProcessDirectives bool
```

Comment:
I can imagine that this will also control the `setOrder` parallel list described in https://github.com/kubernetes/community/pull/537. So I change the name to make it more general.
Will update the comment.